### PR TITLE
Fix fr_channel_drain so that have_read counter gets updated when read…

### PR DESCRIPTION
**Pull request by pwdng**
… stops after header

### CODE REVIEW

File: src/main/command.c [hunk 1]
```diff
@@ -234,8 +234,8 @@ static int fr_server_domain_socket(char const *path, gid_t gid)
 		fr_strerror_printf("Failed determining parent directory");
 	error:
 		talloc_free(dir);
-		if (dir_fd >= 0) close(dir_fd);
-		if (path_fd >= 0) close(path_fd);
+		close(dir_fd);
+		close(path_fd);
 		if (sock_fd >= 0) close(sock_fd);
```
Comment (alandekok) [inline src/main/command.c]: Why this change?

**Comment by pwdng** [comment]
Oops, that's a mistake. I'll remove it.
